Mathlib/Analysis/InnerProductSpace/Adjoint.lean

```diff
@@ -308,18 +308,36 @@ theorem conj_starProjection {T : E →L[𝕜] E} (hT : IsSelfAdjoint T)
 
 end IsSelfAdjoint
 
+namespace ContinuousLinearMap
+
+variable {T : E →L[𝕜] E} [CompleteSpace E]
+
 open ContinuousLinearMap in
-theorem IsIdempotentElem.hasOrthogonalProjection_range [CompleteSpace E]
-    {p : E →L[𝕜] E} (hp : IsIdempotentElem p) : (LinearMap.range p).HasOrthogonalProjection :=
-  have := hp.isClosed_range.completeSpace_coe
-  .ofCompleteSpace _
+theorem IsIdempotentElem.adjoint_range_eq_range_of (hT : IsIdempotentElem T)
+    (h : (LinearMap.range T)ᗮ = LinearMap.ker T) :
+    LinearMap.range (adjoint T) = LinearMap.range T := by
+  have := hT.hasOrthogonalProjection_range
+  have := hT.star.hasOrthogonalProjection_range
+  rw [← Submodule.orthogonal_orthogonal (LinearMap.range (adjoint T)),
+    orthogonal_range, adjoint_adjoint, ← h, Submodule.orthogonal_orthogonal]
```

Comment:
I would omit this. The hypothesis (using the next theorem trivially) can be used to show that `adjoint T = T`, at which point this result is instantaneous.